cg/utils/commands.py

```diff
@@ -0,0 +1,96 @@
+"""
+Code to handle communications to the shell from CG
+"""
+
+import copy
+import logging
+import subprocess
+from subprocess import CalledProcessError
+
+LOG = logging.getLogger(__name__)
+
+
+class Process:
+    """Class to handle communication with other programs via the shell
+
+    The other parts of the code should not need to have any knowledge about how the processes are
+    called, that will be handled in this module.Output form stdout and stdin will be handeld here.
+    """
+
+    def __init__(self, binary, config=None, config_parameter="--config"):
+        """
+        Args:
+            binary(str): Path to binary for the process to use
+            config(str): Path to config if used by process
+        """
+        super(Process, self).__init__()
+        self.binary = binary
+        LOG.info("Initialising Process with binary: %s", self.binary)
+        self.base_call = [self.binary]
+        if config:
+            self.base_call.extend([config_parameter, config])
+        LOG.info("Use base call %s", self.base_call)
+        self._stdout = ""
+        self._stderr = ""
+
+    def run_command(self, parameters=None):
+        """Execute a command in the shell
+
+        Args:
+            parameters(list)
+        """
+        command = copy.deepcopy(self.base_call)
+        if parameters:
+            command.extend(parameters)
+        try:
+            LOG.info("Running command %s", command)
+            res = subprocess.run(
+                command, check=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE
+            )
+        except CalledProcessError as err:
+            LOG.warning(err)
+            raise err
+
+        self.stdout = res.stdout.decode("utf-8").rstrip()
```

Comment:
Here stdout and stderr will always be the same. Did you mean to set 
```
self.stderr = res.stderr.decode("utf-8").rstrip()
```

Reply:
Yes Thank you! Need to add a test for this as well